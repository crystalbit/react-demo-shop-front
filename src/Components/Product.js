import React/*, { useState, useEffect }*/ from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import { red } from '@material-ui/core/colors';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ShareIcon from '@material-ui/icons/Share';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import config from '../config';
import path from 'path';

const useStyles = makeStyles(theme => ({
    card: {
      width: 280,
    },
    media: {
      height: 0,
      paddingTop: '56.25%', // 16:9
    },
    avatar: {
      backgroundColor: red[500],
    },
}));

function Product(props) {
    const classes = useStyles();

    return (
        <Card className={classes.card}>
            <CardHeader
                avatar={
                    <Avatar aria-label="recipe" className={classes.avatar}>
                        R
                    </Avatar>
                }
                title={props.product.name}
                subheader="September 14, 2016"
            />
      <CardMedia
        className={classes.media}
        image={path.join(config.directories.images, props.product.image || '')}
        title={props.product.name}
      />
      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
          {props.product.description}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <IconButton aria-label="add to favorites">
          <FavoriteIcon />
        </IconButton>
        <IconButton aria-label="share">
          <ShareIcon />
        </IconButton>
        <IconButton
          className={classes.expand}
          aria-label="show more"
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
    </Card>
    );
}

export default Product;
