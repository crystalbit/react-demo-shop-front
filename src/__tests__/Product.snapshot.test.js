import React from 'react';
import renderer from 'react-test-renderer';
import Product from '../Components/Product';

test('Product equals its snaphot', () => {
  const component = renderer.create(
    <Product
      product={{
        id: 1,
        name: 'A Product',
        subheader: 'sub',
        image: 'california.jpg',
        description: 'good product',
        price: 100
      }}
      quantity={5}
      onAdd={console.log}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
