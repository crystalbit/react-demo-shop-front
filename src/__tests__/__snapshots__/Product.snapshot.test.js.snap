// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product equals its snaphot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        A Product
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        ðŸŒ sub ðŸŒ 
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root makeStyles-media-2"
    style={
      Object {
        "backgroundImage": "url(\\"/uploads/california.jpg\\")",
      }
    }
    title="A Product"
  />
  <div
    className="MuiCardContent-root makeStyles-content-3"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      good product
    </p>
  </div>
  <div
    className="MuiCardActions-root makeStyles-actions-4"
  >
    <p
      className="MuiTypography-root makeStyles-actionsTypography-5 MuiTypography-h5 MuiTypography-colorTextPrimary"
    >
      $
      100
    </p>
    <span
      className="MuiBadge-root"
    >
      <div
        className="MuiButtonBase-root MuiChip-root MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-deletableColorPrimary MuiChip-clickable MuiChip-deletable"
        mr={4}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiChip-icon MuiChip-iconColorPrimary"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <span
          className="MuiChip-label"
        >
          Add to cart
        </span>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiChip-deleteIcon MuiChip-deleteIconColorPrimary"
          focusable="false"
          onClick={[Function]}
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
          />
        </svg>
      </div>
      <span
        className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
      >
        5
      </span>
    </span>
  </div>
</div>
`;
